src/sentry/tasks/reprocessing2.py

```diff
@@ -16,34 +15,50 @@
     time_limit=120,
     soft_time_limit=110,
 )
-def reprocess_group(project_id, group_id, offset=0, start_time=None):
+def reprocess_group(project_id, group_id, offset=0, start_time=None, max_events=None):
+    from sentry.reprocessing2 import start_group_reprocessing
+
+    start_group_reprocessing(project_id, group_id, max_events=max_events)
+
     if start_time is None:
         start_time = time.time()
 
-    events = list(
-        eventstore.get_unfetched_events(
-            eventstore.Filter(
-                project_ids=[project_id],
-                group_ids=[group_id],
-                # XXX: received?
-                conditions=[["timestamp", "<", to_datetime(start_time)]],
-            ),
-            limit=GROUP_REPROCESSING_CHUNK_SIZE,
-            offset=offset,
-            referrer="reprocessing2.reprocess_group",
+    if max_events is not None and max_events <= 0:
+        events = []
+    else:
+        limit = GROUP_REPROCESSING_CHUNK_SIZE
+
+        if max_events is not None:
+            limit = min(limit, max_events)
+
+        events = list(
```

Comment:
paginated snuba query via offset, from what I can tell this is similar to csv export

Reply:
Logically this seems fine to me, I'm not sure if @fpacifici has any additional thoughts about query performance.